core/che-core-db/src/main/java/org/eclipse/che/core/db/JNDIDataSourceFactory.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.core.db;
+
+
+import org.apache.tomcat.dbcp.dbcp2.BasicDataSource;
+import org.apache.tomcat.dbcp.dbcp2.BasicDataSourceFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.spi.ObjectFactory;
+import java.util.Hashtable;
+import java.util.Properties;
+
+public class JNDIDataSourceFactory implements ObjectFactory {
```

Comment:
AFAIK this is an old version of pool that comes from 
```
        <dependency>
            <groupId>org.apache.tomcat</groupId>
            <artifactId>tomcat-jdbc</artifactId>
            <version>${org.apache.tomcat.version}</version>
        </dependency>
```
But I would like to use the second version of pool  https://tomcat.apache.org/tomcat-8.0-doc/jndi-datasource-examples-howto.html#Database_Connection_Pool_(DBCP_2)_Configurations
```
        <dependency>
            <groupId>org.apache.tomcat</groupId>
            <artifactId>tomcat-dbcp</artifactId>
            <scope>provided</scope>
        </dependency>
```
The main reason why do I need custom factory is the way how I would like to configure it. Tomcat's factories are made to be configured with standard J2EE mechanism aka context.xml, but I would like to use environment variables. With @riuvshin  we considered one more option. Like: using system variables to configure DS in context.xml or server.xml. But found that we print all system variables in tomcat startup, and printing all DS configuration including password doesn't look good. So we decided that custom  ObjectFactory configured over environment variables is a suitable wa